### ISSUE TITLE
directionality of antennas

### ISSUE THREAD

**Issue by kaurss** [nyuwireless/ns3-mmwave#14]
Hi,
I am working on a scenario where I want to check the directionality of the antenna placed on the base station. For this I want to check the number of packets being received by user and see how the number of packets received vary with direction. The problem is I am not sure how to use the antenna-array model and combine it with mmwave-spectrum-phy
Or is my approach totally wrong for the scenario I'm trying to build.
Please help.Thank you.

**Comment by MengleiZhang** [comment]
Hi SUKHMANK,

In this version, the BS/UE antenna panel fixed on the y-z axis, thus rotation is not modeled. 

If you are talking about the how the direction of the beams, which is formed by the antenna arrays applying beamforming vector, affects the throughput. You may try to go through the SellScan method in mmwave-3gpp-channel.cc <http://mmwave-3gpp-channel.cc/>, where the direction of the antenna beams is changed by calling method SetSector(secotorID, antennaNum, elevationAngle). You may repeat the experiment multiple time with different sectorID and elevationAngle. The physical throughput can be computed from the RxPacketTrace.txt. There is a column recording the transport block size, which is number of Bytes transmitted in one TTI(125us).

Menglei



Menglei

**Comment by SUKHMANK** [comment]
Hi,
Thanks for the quick response.
But I'm trying to look at the packets received by the user.

**Comment by MengleiZhang** [comment]
Hi,

Take a look the mmwave-tcp-multiflow.cc example file. It generates a trace
file to record the arrival time and size in Bytes of each packet, with the
following command,

sinkApps.Get(i+nodeNum)->TraceConnectWithoutContext("Rx",MakeBoundCallback
(&Rx, stream));

Menglei

**Comment by navjotsingh963** [comment]
Hi,
I am working on directional antennas. How I can setup directional antennas using mm-wave ns3 module, I looked into your mmwave-3gpp-channel.cc,  I got that. But how I can create my own ns3 program using that method. I want to implement directional antenna working in sector 1 and not it should not be working in other sectors.
Is that possible to implement directional antennas using the mm-wave-3gpp module?
Thank you very much
Navjot Singh
